src/orange/detail/LevelStateAccessor.hh

```diff
@@ -0,0 +1,152 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021-2022 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/LevelStateAccessor.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/data/Collection.hh"
+#include "orange/OrangeData.hh"
+#include "orange/OrangeTypes.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Accesss the 2D fields (i.e., {thread, level}) of OrangeStateData
+ */
+class LevelStateAccessor
+{
+  public:
+    //!@{
+    //! Type aliases
+    using StateRef = NativeRef<OrangeStateData>;
+    //!@}
+
+  public:
+    // Construct from states and indices
+    explicit inline CELER_FUNCTION LevelStateAccessor(StateRef const* states,
```

Comment:
This doesn't need to be (shouldn't be) explicit since it has more than one argument.